### PR TITLE
POLIO-658: create widget for budget timelines frontend

### PR DESCRIPTION
**Pull request by codeKameleon**
Explain what problem this PR is resolving

## Self proof reading checklist

- [x] Did I use eslint and black formatters
- [x] Is my code clear enough and well documented
- [x] Are my typescript files well typed
- [ ] Did I add translations
- [ ] My migrations file are included
- [ ] Are there enough tests
## Changes

- create new component `BudgetTimeline` (use of Stepper material ui component)
- Add BudgetTimeline in `BudgetDetails`

## How to test

1. Make sure you have a polio user and the [config](https://drive.google.com/file/d/1SxAVd1uWSzLIqsPzf2Zi4TkU8Kuv51VQ/view?usp=sharing)* to test budget workflow : [POLIO BUDGET WORKFLOW](https://wiki.bluesquare.org/en/Dev-team/product-management/iaso/polio_budget_workflow)
2. Go to Polio > Budget >  Choose a row and click on Budget details 

 ### If you already have a polio account and user:
 Go to Django Admin > Workflow models > Then, add a new model with this config*

### If you don't have a polio account and user : 

[Set up locally via fixtures](https://wiki.bluesquare.org/en/Dev-team/product-management/iaso/polio_budget_workflow#set-up-locally-via-fixtures)

## Print screen / video
Capture d’écran du 2022-11-03 17-40-59

## Note : 

This [PR](https://github.com/BLSQ/bluesquare-components/pull/89) needs to be merged to see the light green from `stepActive` css class

### CODE REVIEW

File: plugins/polio/js/src/pages/Budget/hooks/api/useGetBudget.ts [hunk 1]
```diff
@@ -72,12 +72,22 @@ export const useGetBudgetForCampaign = (
...
     };
+
     return useSnackQuery({
         queryFn: () => getBudgetForCampaign(id, params),
         queryKey: ['budget', 'campaign', id],
-        options: { enabled: Boolean(id) },
+        options: {
+            enabled: Boolean(id),
+            select: (data: Partial<Budget>) => {
+                const newData = {
```
Comment (quang-le) [inline plugins/polio/js/src/pages/Budget/hooks/api/useGetBudget.ts]: what does that do?

File: plugins/polio/js/src/pages/Budget/BudgetDetails/BudgetTimeline.tsx [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        <>
+            <Stepper
+                classes={classes}
+                activeStep={activeStep}
+                alternativeLabel
+                orientation={orientation}
+            >
+                {categories?.map(category => (
+                    <Step
+                        style={{
```
Comment (quang-le) [inline plugins/polio/js/src/pages/Budget/BudgetDetails/BudgetTimeline.tsx]: use a classname instead

File: plugins/polio/js/src/pages/Budget/BudgetDetails/BudgetTimeline.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+                                        ) : (
+                                            <CheckBoxOutlineBlank />
+                                        )}
+                                        <div>{item.label}</div>
+                                    </div>
+
+                                    {item.performed_at && (
+                                        <div
+                                            style={{
+                                                display: 'flex',
```
Comment (quang-le) [inline plugins/polio/js/src/pages/Budget/BudgetDetails/BudgetTimeline.tsx]: idem

File: plugins/polio/js/src/pages/Budget/BudgetDetails/BudgetTimeline.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                            {category.items.map(item => (
+                                <Box
+                                    display="flex"
+                                    alignItems="center"
+                                    justifyContent="flex-start"
+                                    flexDirection="column"
+                                >
+                                    <div
+                                        style={{
+                                            display: 'flex',
```
Comment (quang-le) [inline plugins/polio/js/src/pages/Budget/BudgetDetails/BudgetTimeline.tsx]: idem

File: plugins/polio/js/src/styles/constants.js [hunk 5]
```diff
@@ -2,6 +2,10 @@ export const OK_COLOR = 'green';
 export const FAIL_COLOR = 'red';
 export const WARNING_COLOR = '#FFD835';
 
+export const BUDGET_COMPLETED_COLOR = '#008000';
+export const BUDGET_PENDING_COLOR = '#90EE90';
+export const BUDGET_INACTIVE_COLOR = 'grey';
```
Comment (quang-le) [inline plugins/polio/js/src/styles/constants.js]: maybe get the gray from the theme

**Review by quang-le — CHANGES_REQUESTED** [review]

**Comment by codeKameleon** [comment]
@cypress

**Comment by quang-le** [comment]
The json you link to is invalid

**Review by quang-le — CHANGES_REQUESTED** [review]
The json you link to has been made from an out of date version of the workflow. Please take the latest version and update it

**Review by quang-le — CHANGES_REQUESTED** [review]
The code is pretty clean, nice work.
In hindsight, greying the icon is confusing when you start a new budget flow. It looks a bit as if the whole process as been completed:

Screenshot 2022-11-07 at 09 54 51

There seems to be a config error as "WHO sent budget" is never marked as completed. Please make sure to use a copy of the workflow used in campaigns-staging:

Screenshot 2022-11-07 at 09 30 47

There's not enough contrast between the colors for active and inactive. I know I suggested to use the gray from the theme. It was a bad idea:

Screenshot 2022-11-07 at 09 31 20


Step 2 should appear as completed. It doesn't. I don't know if it's a config problem or a problem with the component:

Screenshot 2022-11-07 at 09 32 03

The console is full of errors, please clean it up:

Screenshot 2022-11-07 at 09 34 39

And as mentioned before: please make sure to update a copy of the workflow used in campaigns-staging with your changes, or we won't be able to show the changes to the client

**Comment by beygorghor** [comment]
I have fixed errors and change the lay-out a bit
Screenshot 2022-11-08 at 10 14 50
Screenshot 2022-11-08 at 10 14 32

**Comment by codeKameleon** [comment]
switch the PR to Draft because of the budget worflow approval issues : https://bluesquare.atlassian.net/browse/POLIO-681

**Comment by beygorghor** [comment]
@codeKameleon just hide he widget for now

**Pull request merged**